### PR TITLE
Explained the difference between ownership iteration and reference iteration

### PR DESCRIPTION
**Pull request by srinivasreddy**
Issue here : https://github.com/rust-lang/rust/issues/31991

### CODE REVIEW

File: src/doc/book/vectors.md [hunk 1]
```diff
@@ -114,7 +114,30 @@ for i in v {
     println!("Take ownership of the vector and its element {}", i);
 }
 ```
+Note: You cannot use the vector again once you have iterated with ownership of the vector.
```
Comment (Manishearth) [inline src/doc/book/vectors.md]: iterated by taking ownership of the vector

File: src/doc/book/vectors.md [hunk 2]
```diff
@@ -114,7 +114,30 @@ for i in v {
...
+You can iterate the vector multiple times with reference iteration. For example, the following
```
Comment (Manishearth) [inline src/doc/book/vectors.md]: by taking a reference to the vector whilst iterating

File: src/doc/book/vectors.md [hunk 3]
```diff
@@ -114,7 +114,30 @@ for i in v {
...
+for i in v {
+    println!("Take ownership of the vector and its element {}", i);
+}
+```
+Whereas the following works perfectly,
 
+```rust
+let mut v = vec![1, 2, 3, 4, 5];
+for i in &v {
+    println!("A mutable reference to {}", i);
```
Comment (Manishearth) [inline src/doc/book/vectors.md]: it's not mutable in this case. Just say "This is a reference"

File: src/doc/book/vectors.md (hunk 1 above)
Comment (steveklabnik) [inline src/doc/book/vectors.md]: (Also, please put a newline above this line as well)

File: src/doc/book/vectors.md [hunk 4]
```diff
@@ -114,7 +114,30 @@ for i in v {
     println!("Take ownership of the vector and its element {}", i);
 }
 ```
+Note: You cannot use the vector again once you have iterated with ownership of the vector.
+You can iterate the vector multiple times with reference iteration. For example, the following
+code does not compile.
+```rust
```
Comment (steveklabnik) [inline src/doc/book/vectors.md]: Blank line above here please!

File: src/doc/book/vectors.md [hunk 5]
```diff
@@ -114,7 +114,30 @@ for i in v {
...
+let mut v = vec![1, 2, 3, 4, 5];
+for i in v {
+    println!("Take ownership of the vector and its element {}", i);
+}
+for i in v {
+    println!("Take ownership of the vector and its element {}", i);
+}
+```
+Whereas the following works perfectly,
```
Comment (steveklabnik) [inline src/doc/book/vectors.md]: and above here

File: src/doc/book/vectors.md [hunk 6]
```diff
@@ -114,7 +114,30 @@ for i in v {
...
+```rust
+let mut v = vec![1, 2, 3, 4, 5];
+for i in &v {
+    println!("A mutable reference to {}", i);
+}
+
+for i in &v {
+    println!("A mutable reference to {}", i);
+}
+```
```
Comment (steveklabnik) [inline src/doc/book/vectors.md]: and below here

File: src/doc/book/vectors.md [hunk 7]
```diff
@@ -114,7 +114,30 @@ for i in v {
     println!("Take ownership of the vector and its element {}", i);
 }
 ```
+Note: You cannot use the vector again once you have iterated with ownership of the vector.
+You can iterate the vector multiple times with reference iteration. For example, the following
+code does not compile.
+```rust
+let mut v = vec![1, 2, 3, 4, 5];
+for i in v {
```
Comment (steveklabnik) [inline src/doc/book/vectors.md]: and between this let and for

File: src/doc/book/vectors.md [hunk 8]
```diff
@@ -114,7 +114,30 @@ for i in v {
...
+    println!("Take ownership of the vector and its element {}", i);
+}
+for i in v {
```
Comment (steveklabnik) [inline src/doc/book/vectors.md]: and between these two fors

File: src/doc/book/vectors.md [hunk 9]
```diff
@@ -114,7 +114,30 @@ for i in v {
...
+    println!("Take ownership of the vector and its element {}", i);
+}
+```
+Whereas the following works perfectly,
 
+```rust
+let mut v = vec![1, 2, 3, 4, 5];
+for i in &v {
```
Comment (steveklabnik) [inline src/doc/book/vectors.md]: and between this let and for

**Comment by steveklabnik** [comment]
Thanks so much @srinivasreddy ! I like @Manishearth 's grammar suggestions, and I have some whitespace ones as well. We're good to go after that's fixed up!

**Comment by srinivasreddy** [comment]
@steveklabnik  Welcome!!  @Manishearth  Please review and merge

File: src/doc/book/vectors.md [hunk 10]
```diff
@@ -115,6 +115,36 @@ for i in v {
...
+
+```rust
+let mut v = vec![1, 2, 3, 4, 5];
+
+for i in &v {
+    println!("This is a reference to {}", i);
+}
+
+for i in &v {
+    println!("This is a reference {}", i);
```
Comment (Manishearth) [inline src/doc/book/vectors.md]: make this text match the previous

**Comment by srinivasreddy** [comment]
@Manishearth Done

**Comment by Manishearth** [comment]
Thanks!

**Comment by srinivasreddy** [comment]
@Manishearth I am new to travis. Do you see any reason for build failure? Could you please help me ?

File: src/doc/book/vectors.md [hunk 11]
```diff
@@ -115,6 +115,36 @@ for i in v {
 }
 ```
 
+Note: You cannot use the vector again once you have iterated by taking ownership of the vector.
+You can iterate the vector multiple times by taking a reference to the vector whilst iterating. 
+For example, the following code does not compile.
+
+```rust
```
Comment (Manishearth) [inline src/doc/book/vectors.md]: you need to have `rust,ignore` here

**Comment by Manishearth** [comment]
Ah, in this case the doctest failed. Rust code in documentation is compiled and run as a test, so your example of failing code should be annotated as mentioned above to skip this.

File: src/doc/book/vectors.md (hunk 11 above)
Comment (srinivasreddy) [inline src/doc/book/vectors.md]: @Manishearth Done. Please issue auto merge

**Comment by srinivasreddy** [comment]
@Manishearth Please merge. Or  could  @bors automatically merge this?

**Comment by Manishearth** [comment]
Yes. Bors will run tests on the CI and automatically merge if they pass, like last time.

**Comment by Manishearth** [comment]
There's a queue (http://buildbot.rust-lang.org/homu/queue/rust), so it might take some time for your PR to get merged.

**Pull request merged**